addons/website_quote/controllers/main.py

```diff
@@ -160,18 +145,13 @@ def add(self, option_id, order_id, token, **post):
             'discount': option.discount,
         }
 
-        line = request.registry.get('sale.order.line').create(request.cr, SUPERUSER_ID, vals, context=request.context)
-        request.registry.get('sale.order.line')._compute_tax_id(request.cr, SUPERUSER_ID, [line], context=request.context)
-        option_obj.write(request.cr, SUPERUSER_ID, [option.id], {'line_id': line}, context=request.context)
+        order_line = request.env['sale.order.line'].sudo().create(vals)
+        order_line._compute_tax_id()
+        option.line_id = order_line
         return werkzeug.utils.redirect("/quote/%s/%s#pricing" % (order.id, token))
 
-    # note dbo: website_sale code
-    @http.route(['/quote/<int:order_id>/transaction/<int:acquirer_id>'], type='json', auth="public", website=True)
-    def payment_transaction(self, acquirer_id, order_id):
-        return self.payment_transaction_token(acquirer_id, order_id, None)
-
     @http.route(['/quote/<int:order_id>/transaction/<int:acquirer_id>/<token>'], type='json', auth="public", website=True)
-    def payment_transaction_token(self, acquirer_id, order_id, token):
+    def payment_transaction_token(self, acquirer_id, order_id, token=None):
```

Comment:
Ok I understand, and it makes sence. But :
- put it in separated commit (migrate both method, and remove one later)
- your route will not match since its path requires `token`.
- Maybe we should, in the javascript where the route is called, put the token in its param dict, and remove it from the route path in controller. 
I'm not totally sure, you should ask to somebody else.